src/middleware/withEgressClient.js

```diff
@@ -89,8 +89,10 @@ async function record (space, resource, bytes, servedAt, env, ctx) {
     nb: {
       resource,
       bytes,
-      servedAt: Math.floor(servedAt.getTime() / 1000)
+      servedAt: servedAt.getTime()
     },
+    expiration: Infinity, // Don't expire the invocation, so we can record egress any time
```

Comment:
I'm not sure this is necessary, and I don't think the upload service should be able to record egress for this resource forever...just when it receives this invocation.

Reply:
I wanted to include this here explicitly because I was thinking about the rollout for Phase 2 (https://github.com/storacha/project-tracking/issues/192). In this phase, I plan to integrate a queue in Cloudflare worker to receive the egress events, with consumers responsible for signing and forwarding these requests to the Upload-API. This setup would allow us to fine-tune the duration for which these events are recorded, likely based on the throughput of the consumers. Also, this code would probably be moved to the consumer.